cpp/src/reductions/simple.cuh

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "reduction.cuh"
+#include "reduction_operators.cuh"
+
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+
+namespace cudf {
+namespace experimental {
+namespace reduction {
+namespace simple {
+
+/** --------------------------------------------------------------------------*    
+ * @brief Reduction for 'sum', 'product', 'min', 'max', 'sum of squares'
+ * which directly compute the reduction by a single step reduction call
+ *
+ * @param[in] col    input column view
+ * @param[out] scalar  output scalar data
+ * @param[in] stream cuda stream
+ *
+ * @tparam ElementType  the input column cudf dtype
+ * @tparam ResultType   the output cudf dtype
+ * @tparam Op           the operator of cudf::experimental::reduction::op::
+ * ----------------------------------------------------------------------------**/
+template<typename ElementType, typename ResultType, typename Op>
+std::unique_ptr<scalar> simple_reduction(column_view const& col, data_type const output_dtype, cudaStream_t stream,
+    rmm::mr::device_memory_resource* mr )
+{
+    std::unique_ptr<scalar> result;
+    if(std::is_same<string_scalar::value_type, ResultType>::value) {
+      //TODO cudf::string_view support 
+      result = make_string_scalar("min/max/sum", stream, mr);
```

Comment:
What is this? "min/max/sum"????

Reply:
`//TODO cudf::string_view support `
I am developing template specialization for `string_view` because `string_scalar` is read only and interfaces are different too.

Reply:
**Another issue for `string_view`**
for operator `max`, smallest value `string_view{nullptr, 0}` works. 
for operator `min`, there is no largest `string_view` value to use in null_replacement_iterator.

Reply:
removed it. added min, max support for string_view